<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Python Tutorial  - 16. Exception Handling | Coder Coacher - Coaching Coders</title><meta content="Python Tutorial  - 16. Exception Handling - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/codebasics/">codebasics</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Python Tutorial  - 16. Exception Handling</b></h2><h5 class="post__date">2016-02-27</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/Ia1i5EIGp9k" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">hello and welcome to coal basics coding
tutorials today we are going to talk
about exceptions and here is the list of
topics that we are covering in this
video okay let's begin with what are
exception exceptions are the errors that
occurs while executing your program
let's say you are driving on a road and
the road is clear you reach your
destination safely without any trouble
this is called executing program without
any exception but life is not same all
it is on one fine day when you are
driving on a road you see a scene like
this it's quite funny but things like
this happen and what you need to do is
take a detour so the accident that you
saw on a row is called
exception because you didn't accept
expected it to happen but it did and
your action of taking a detour is called
handling exception okay so see these
kind of accidents happen when you're
writing code as well so let's go over
some of the basic exceptions I'm going
to bring up idle here and we will see
some basic exit exceptions
the first one is dividing a number by
zero when you do that you get zero
division error exception because you
know you cannot divide number by zero
you basically get infinity another one
is when you try to concatenate string
with a number it says can't convert
integer object to string explicitly
whenever you see this kind of trace back
it very first generally to an exception
okay and whenever exception occurs your
program terminates execution this is
called crash your program crashed okay
now we are going to write a program in
PH arm to show you how to handle
exceptions this program is very simple
it takes two numbers as an input in fact
what I'm going to do is just copy my
code to save you time on recording as
you see here all we are doing is taking
two inputs from a user and dividing that
number and printing the division okay if
you execute this program and you enter
normal two numbers it works fine it says
my division is 2.0 as you see here okay
but what happens when you have scenario
like this again you are dividing a
number by 0 and you get a crash now
interesting thing to notice here is when
the crash happened
it stopped execution of program at this
point you do not see a print statement
like this bit spin statement being
executed right now if you are writing a
huge program plus a thousand line
program and this kind of situation
happens in the middle of your program
will terminate the execution in the
middle which is not good what you want
to do is you want to handle the
exception it is like you are driving on
a road you see an accident and you just
drive back home you generally don't do
that right you don't exit that terminate
your plan in between you find
alternative way which is a detour and
you still reach the destination same
thing should happen while writing a
program as well and for doing that you
have to handle the exception and the way
you handle exception
is by writing try-catch so you said die
and then here you will say accept
so try Colin and the program the the
block of program that you expect could
possibly generate exception you should
put it on ink like except block so with
in between try and except you should
have that core okay so the syntax here
is except exception as II okay and you
will say okay and exception occurred and
you want to just lets first print
exception and see what happens when you
run this program I'm entering the number
and the same names is not defined okay
this was another problem so okay what
I'm doing is define a kernel divided
number by 0 I'm initializing Z as none
when you do this see here interesting
thing to notice is it executed a program
without any crash you did not see those
red lines here you are seeing division
is whatever right so if you have let's
say 100 lines of code returned below
this line it will all get executed if
you handle exception this way okay so
this program now is pretty stable
it works with normal scenarios where the
exception don't happen it also works
with the scenarios where exceptions do
happen so this program New York never
crashes so that's a benefit of handling
an exception now here I am handling an
exception which is a very generic
exception on what you want to do is you
want to handle a very specific exception
so for example this exception is
actually called zero division error okay
and when that happens you can directly
write division by zero print it this way
you you expect specific situation and
you handle that specific situation it is
not a generic way of handling things
okay if you have to compare this code
with our real-life example of accident
and d2 then the code will look like this
you're driving you encounter like an
exception of accident and the way you
handle it is by taking a detour okay
let's now talk about how to handle
multiple exception this program is on
already handling zero division error
exception
but let's say by mistake when you take
the number from the console you forget
to convert it to an integer so now what
I'm doing is I'm dividing this X will be
a string by the way when you enter
something from the console it is by
default a string which you need to
convert it to a and int but let's say
you don't do that let's see what happens
okay so I'll just enter normal numbers
here Wow
you saw it crashed because you cannot
divide string with integer so what
should we do here first of all if you
want to handle this exception you want
to figure out which exception type that
is okay so you first right so let's
first cover how to figure out the type
of exception you will say you just
handle it as a generic exception first
of all because you don't know the exact
name right okay so that's why you will
say print except exception type and my
exception type here would be
this is how you figure out the type of
exception dot underscore underscore name
okay let's run this again you're getting
the same thing so you have to always say
this I'm going to move this on the right
hand side the way you move this on the
right hand side is you will say okay
it's probably not visible so I'm going
to minimize that this make this getting
smaller
you'll see move to right okay so now you
can see what's being executed all right
when we run this program and you ain't
ha ha
now the same exception type is type
error so you know what is the exception
type so let's put type error here and
here you can see type error exception
okay when you run this is a 40 I read to
you get type error exception if you run
this and then upon noticing this
exception obviously you will examine
your code you will fix it and then when
you run it again it just starts working
fine also if you have / 0 exception then
also it will handle it so this will
pretty much handle all kinds of cases so
that was the basic intro on exception
thanks for what</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>